optimum/neuron/modeling_decoder.py

```diff
@@ -224,3 +236,45 @@ def _save_pretrained(self, save_directory: Union[str, Path]):
             self.model_path = save_directory
 
         self.generation_config.save_pretrained(save_directory)
+
+    def push_to_hub(
+        self,
+        save_directory: str,
+        repository_id: str,
+        private: Optional[bool] = None,
+        use_auth_token: Union[bool, str] = True,
+        endpoint: Optional[str] = None,
+    ) -> str:
+        if isinstance(use_auth_token, str):
+            huggingface_token = use_auth_token
+        elif use_auth_token:
+            huggingface_token = HfFolder.get_token()
+        else:
+            raise ValueError("You need to provide `use_auth_token` to be able to push to the hub")
+        api = HfApi(endpoint=endpoint)
+
+        user = api.whoami(huggingface_token)
+        self.git_config_username_and_email(git_email=user["email"], git_user=user["fullname"])
```

Comment:
Why?

Reply:
This is a copy/paste from optimum. I assume it allows to identify the author if it is not the same as the one identified by the token (same thing ads when you add a commit from someone else in your branch/pull-request).